lib/livebook_web/live/hooks/app_auth_hook.ex

```diff
@@ -73,9 +95,11 @@ defmodule LivebookWeb.AppAuthHook do
     {:cont, socket}
   end
 
-  defp livebook_authenticated?(session, socket) do
+  defp livebook_authenticated?(session, user, socket) do
     uri = get_connect_info(socket, :uri)
-    LivebookWeb.AuthPlug.authenticated?(session, uri.port)
+
+    LivebookWeb.AuthPlug.authenticated?(session, uri.port) and
+      user.restricted_apps_groups == nil
```

Comment:
So far, for the Livebook admin part, authenticated and authorized was he same. The comment at the top says:

```
  #   * `:livebook_authenticated?` - if the user has full Livebook
  #     access
```

So I think now we need to adjust those wordings to say authorized instead.

We can have `LivebookWeb.AuthPlug.authorized?` and make the `user.restricted_apps_groups == nil` check there. The assign should also become `livebook_authorized?` (you can grep it, it's just a few occurrences).

Does that make sense?